hydrolib/profile_optimizer/profile_optimizer/geometry.py

```diff
@@ -0,0 +1,153 @@
+import pandas as pd
+import geopandas as gpd
+import xarray as xr
+from shapely.geometry import LineString
+from shapely.errors import ShapelyDeprecationWarning
+from pathlib import Path
+import warnings
+from hydrolib.core.io.crosssection.models import CrossLocModel
+
```

Comment:
Added some easily added type hints. Due to parallel development at RHDHV I'm however not going to make too big changes, because there are bigger improvements in the pipeline.